Camera branch (#17394)
zephyrproject-rtos/zephyr: samples/camera/README.rst

.. camera:

Camera Sample Application
###################################

Overview
********
This project demos camera samples.
It should work with any platform featuring a I2C peripheral interface and a CMOS sensor interface.
It does not work on QEMU.

> If there is display device on board, the demo displays camera stream on the display device.

Reviewer: ```suggestion
If there is a display device on the board, the demo displays the camera stream on the display device.
```

Author: Fixed.

Requirements
************

> This project requires a CMOS sensor connected to board.

Reviewer: Can you list the sensor (and LCD display panel) you tested this with (so someone could duplicate what you did.

Author: Added.

Building and Running
********************
This sample can be built for a ``mimxrt1050_evk`` board with mt9m114 CMOS sensor connected.

.. zephyr-app-commands::
   :zephyr-app: samples/camera
   :board: mimxrt1050_evk
   :goals: build flash

> To run this sample, a CMOS sensor should be present connected to the board.

Reviewer: ```suggestion
To run this sample, a CMOS sensor should be connected to the board.
```

Author: fixed.

The sample outputs the frames per second number on the console in capture
mode and preview mode periodically.

> If there is LCD pannel present on the board, the sample outputs the camera

Reviewer: ```suggestion
If there is an LCD panel present on the board, the sample outputs the camera
```

Author: Fixed.

> If there is LCD pannel present on the board, the sample outputs the camera

Reviewer: You call it a "display device" on line 11, but an "LCD panel" here.  You should pick one term and use it consistently.

Author: Fixed.

> .. _camera:

Reviewer: As written, this seems like a generic camera sample, but I think it's actually a very specific sample for an MT9M114 image sensor connected to a mimxrt1050_evk board.   The documentation should reflect this in the title, description, hardware requirements, and references.

Author: Not only for mt9m114 and mimxrt1050. The sample code uses generic API and no specific low level API used.
But so far, only mt9m114 driver and imxrt CSI driver support on Zephyr.

Author: Not only for mt9m114 and mimxrt1050. The sample code uses generic API and no specific low level API used.
But so far, only mt9m114 driver and imxrt CSI driver support on Zephyr.

Reviewer: OK, thanks for the explanation.

Author: update the README of camera sample

CAMERA: camera
#####################################################

Description
***********

References
**********

Wiring
*******

Building and Running
********************

This project demos camera samples.

> It should work with any platform featuring a I2C peripheral interface and a CMOS sensor interface.

Reviewer: The code seems to be very hardware-specific.  Will it really work with any CMOS sensor and board with an I2C interface?

Author: The sample code and framework are designed for generic camera features. But need more camera drivers and cmos sensors added to test.

Reviewer: OK, thanks for explaining.

It does not work on QEMU.
In this example below the :ref:`mimxrt1050_evk` board is used.

.. zephyr-app-commands::
   :zephyr-app: samples/camera
   :board: mimxrt1050_evk
   :goals: build flash

Sample Output
=============

.. code-block:: console

> <repeats endlessly>

Reviewer: repeats what endlessly?

Author: Update the README.rst
